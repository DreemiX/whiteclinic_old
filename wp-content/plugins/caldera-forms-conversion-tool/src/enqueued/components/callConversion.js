import apiFetch from '@wordpress/api-fetch';
import PropTypes from 'prop-types';

export const callConversion = async ( props ) => {
	apiFetch.use(
		apiFetch.createNonceMiddleware( cf_conversion_tool_vars.rest_nonce )
	);

	try {
		return await apiFetch( {
			path: '/nf-formbuilder/convert-cf',
			method: 'POST',
			data: {
				form: props.cfForm,
			},
			signal: props.signal,
		} );
	} catch ( e ) {
		console.log( e );
	}
};

callConversion.propTypes = {
	cfForm: PropTypes.object.isRequired,
	signal: PropTypes.object.isRequired,
};
