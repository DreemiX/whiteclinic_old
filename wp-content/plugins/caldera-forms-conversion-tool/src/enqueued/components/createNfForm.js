import PropTypes from 'prop-types';
import { Buffer } from 'buffer';

export const createNfForm = async ( props ) => {
	try {
		const { formJson, exportName, handleCreationProcess } = props;
		const data = ';base64,' + Buffer.from( formJson ).toString( 'base64' );

		const formData = {
			name: exportName + '.nff',
			content: data,
		};

		const postData = {
			action: 'nf_batch_process',
			batch_type: 'import_form',
			security: cf_conversion_tool_vars.nf_security_nonce,
			extraData: formData,
		};

		await jQuery.post( ajaxurl, postData ).then( ( response ) => {
			response = JSON.parse( response );
			handleCreationProcess( response );
		} );
	} catch ( e ) {
		console.log( e );
		return e;
	}
};

createNfForm.propTypes = {
	formJson: PropTypes.object.isRequired,
	exportName: PropTypes.string.isRequired,
	signal: PropTypes.object.isRequired,
	handleCreationProcess: PropTypes.func.isRequired,
};
