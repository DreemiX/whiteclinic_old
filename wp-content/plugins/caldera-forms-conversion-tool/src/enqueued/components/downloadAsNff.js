import PropTypes from 'prop-types';

export const downloadAsNff = async ( props ) => {
	let result;
	try {
		const { formJson, exportName } = props;

		const dataStr =
			'data:text/json;charset=utf-8,' + encodeURIComponent( formJson );
		const downloadAnchorNode = document.createElement( 'a' );
		downloadAnchorNode.setAttribute( 'href', dataStr );
		downloadAnchorNode.setAttribute( 'download', exportName + '.nff' );
		document.body.appendChild( downloadAnchorNode );
		downloadAnchorNode.click();
		downloadAnchorNode.remove();

		result = 'success';
	} catch ( e ) {
		result = e;
	}

	return result;
};

downloadAsNff.propTypes = {
	formJson: PropTypes.object.isRequired,
	exportName: PropTypes.string.isRequired,
};
