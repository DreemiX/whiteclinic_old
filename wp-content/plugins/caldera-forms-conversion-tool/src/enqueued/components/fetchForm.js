import apiFetch from '@wordpress/api-fetch';
import PropTypes from 'prop-types';

export const fetchForm = async ( props ) => {
	apiFetch.use(
		apiFetch.createNonceMiddleware( cf_conversion_tool_vars.nonce )
	);

	try {
		return await apiFetch( {
			url:
				window.location.href +
				'&format=json&export-form=' +
				props.formID,
			signal: props.signal,
		} );
	} catch ( e ) {
		console.log( e );
	}
};

fetchForm.propTypes = {
	formID: PropTypes.string.isRequired,
	signal: PropTypes.object.isRequired,
};
