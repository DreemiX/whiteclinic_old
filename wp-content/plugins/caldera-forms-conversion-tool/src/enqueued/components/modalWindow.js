import { Component } from '@wordpress/element';
import PropTypes from 'prop-types';
import { __ } from '@wordpress/i18n';
import { Button, Modal, Spinner } from '@wordpress/components';
import { fetchForm, callConversion, downloadAsNff, createNfForm } from './';

export class ModalWindow extends Component {
	constructor( props ) {
		super( props );
		this.state = {
			modalOpen: false,
			spinnerOn: false,
			createForm: false,
			downloadFile: false,
			getCfFormStep: false,
			getNfFormTranslation: false,
			getNfFormCreationProcess: {},
			getFormResponse: {},
			fetchController: new AbortController(),
		};

		this.handleModal = this.handleModal.bind( this );
		this.conversionAction = this.conversionAction.bind( this );
		this.handleSpinner = this.handleSpinner.bind( this );
		this.handleGetCfFormStep = this.handleGetCfFormStep.bind( this );
		this.handleGetNfFormTranslation = this.handleGetNfFormTranslation.bind(
			this
		);
		this.handleGetNfFormCreationProcess = this.handleGetNfFormCreationProcess.bind(
			this
		);
		this.handleGetFormResponse = this.handleGetFormResponse.bind( this );
		this.cancelAction = this.cancelAction.bind( this );
		this.handleFetchController = this.handleFetchController.bind( this );
	}

	componentWillUnmount() {
		//Make sure we cancel fetch Job when Component gets umounted
		this.state.fetchController.abort();
	}
	//Modal state process
	handleModal() {
		//Refresh response when opening and closing the Modal
		this.handleGetFormResponse( {} );
		this.setState( { modalOpen: ! this.state.modalOpen } );
	}
	//Spinner state process
	handleSpinner( method ) {
		if ( typeof method === 'undefined' ) {
			this.setState( { spinnerOn: ! this.state.spinnerOn } );
		} else {
			let spinner, create, download;
			if ( method === 'off' ) {
				spinner = false;
				create = false;
				download = false;
			} else if ( method === 'create_form' ) {
				spinner = true;
				create = true;
				download = false;
			} else if ( method === 'download_file' ) {
				spinner = true;
				create = false;
				download = true;
			}

			this.setState( {
				spinnerOn: spinner,
				createForm: create,
				downloadFile: download,
			} );
		}
	}
	/**
	 * Sets data form first step of action when CF Form is being retrieved
	 * @param string data message to display during first step of process ( processing/done... )
	 */
	handleGetCfFormStep( data ) {
		this.setState( { getCfFormStep: data } );
	}
	/**
	 * Sets data form second step of action when CF Form is being translated
	 *  @param string data message to display during second step of process ( processing/done... )
	 */
	handleGetNfFormTranslation( data ) {
		this.setState( { getNfFormTranslation: data } );
	}
	/**
	 * Display Message, success  or error
	 * @param response is an object and needs to hold a message entry for it to be printed on the interface
	 */
	handleGetFormResponse( response ) {
		this.setState( { getFormResponse: response } );
	}
	/**
	 * Sets data from Third step step of action when NF Form is being created / downloaded
	 *  @param string data message to display during second step of process ( processing/done... )
	 */
	handleGetNfFormCreationProcess( data ) {
		this.setState( { getNfFormCreationProcess: data } );
	}
	/**
	 * Trigger fetch CF JSON and call translation to NFF
	 * @param string method is "download_file" or "create_form"
	 */
	async conversionAction( method ) {
		//Prepare Component's state
		this.handleGetFormResponse( {} );
		this.handleGetNfFormTranslation( false );
		this.handleGetCfFormStep( false );
		this.handleGetNfFormCreationProcess( {} );
		this.handleSpinner( method );
		//Start process
		try {
			//Start form fetch
			const action =
				method === 'download_file' ? 'download_file' : 'create_form';
			const fetchData = {
				formID: this.props.formID,
				signal: this.state.fetchController.signal,
			};
			this.handleGetCfFormStep(
				__( 'Processing', 'cf_conversion_tool' )
			);
			this.handleGetNfFormTranslation(
				__( 'Waiting for CF Data', 'cf_conversion_tool' )
			);
			this.handleGetNfFormCreationProcess( {
				message: __( 'Waiting for NFF Data', 'cf_conversion_tool' ),
			} );
			//Get CF Form
			const formJson = await fetchForm( fetchData );
			//Display Status
			if ( typeof formJson !== 'undefined' ) {
				this.handleGetCfFormStep( __( 'Done', 'cf_conversion_tool' ) );
				this.handleGetNfFormTranslation(
					__( 'Processing', 'cf_conversion_tool' )
				);
			}
			//Start translation
			const conversionData = {
				cfForm: formJson,
				signal: this.state.fetchController.signal,
			};
			const formConversion = await callConversion( conversionData );

			//Display new Status for Form translation
			if ( typeof formConversion !== 'undefined' ) {
				this.handleGetNfFormTranslation(
					__( 'Done', 'cf_conversion_tool' )
				);
				this.handleGetNfFormCreationProcess( {
					message: __( 'Processing', 'cf_conversion_tool' ),
				} );
				const nffForm = JSON.parse( formConversion.form );
				let result = {
					message: __( 'Unknown Error', 'cf_conversion_tool' ),
				};
				//Trigger File download if it is the chosen method
				if ( action === 'download_file' ) {
					//Trigger Download alert
					const download = await downloadAsNff( {
						formJson: formConversion.form,
						exportName: nffForm.settings.title,
					} );
					//Set result data
					if ( download === 'success' ) {
						result = formConversion;
						result.message = __(
							'Form converted successfully',
							'cf_conversion_tool'
						);
					}
				} else if ( action === 'create_form' ) {
					//Start Form creation with batch import process
					do {
						await createNfForm( {
							formJson: formConversion.form,
							exportName: nffForm.settings.title,
							signal: this.state.fetchController.signal,
							handleCreationProcess: this
								.handleGetNfFormCreationProcess,
						} );
					} while (
						! this.state.getNfFormCreationProcess.batch_complete
					);
					//Set result data
					if ( this.state.getNfFormCreationProcess.batch_complete ) {
						result = this.state.getNfFormCreationProcess;
						result.message = __(
							'Ninja Forms Form created sucessfully',
							'cf_conversion_tool'
						);
						result.action = action;
						result.form = this.state.getNfFormCreationProcess.form_id;
					}
				}

				//Display Success Message
				this.handleGetFormResponse( result );
			}
		} catch ( error ) {
			//Display Errors
			console.log( 'error' + JSON.stringify( error ) );
			this.handleGetFormResponse( error );
		} finally {
			//Stop Spinner and refresh Fetch abortController
			this.handleSpinner( 'off' );
			this.handleFetchController();
		}
	}

	/**
	 * Cancel Fetch Jobs, spinners and reload state
	 */
	cancelAction() {
		this.state.fetchController.abort();
		this.handleGetFormResponse( {} );
		this.handleGetCfFormStep( false );
		this.handleGetNfFormTranslation( false );
		this.handleSpinner( 'off' );
		this.handleFetchController();
	}
	//Refresh Fetch abortController
	handleFetchController() {
		this.setState( { fetchController: new AbortController() } );
	}

	render() {
		const {
			props,
			state,
			handleModal,
			conversionAction,
			cancelAction,
		} = this;
		const {
			modalOpen,
			spinnerOn,
			createForm,
			downloadFile,
			getFormResponse,
			getCfFormStep,
			getNfFormTranslation,
			getNfFormCreationProcess,
		} = state;
		const { formID, form } = props;

		const formText = !form.name ? formID : form.name;
		return (
			<>
				{ ' ' }
				|
				<Button
					isTertiary
					className="cf-converter-modal-action"
					onClick={ handleModal }
					text={ __(
						'Convert to Ninja Forms',
						'cf_conversion_tool'
					) }
				/>
				{ modalOpen && (
					<Modal
						className="cf-converter-modal"
						title={
							__( 'Convert form ', 'cf_conversion_tool' ) +
							formText +
							__(
								' into a Ninja Forms Form.',
								'cf_conversion_tool'
							)
						}
						onRequestClose={ handleModal }
						shouldCloseOnClickOutside={ false }
					>
						<p>
							{ __(
								'To convert this form to a Ninja Forms form, please select ',
								'cf_conversion_tool'
							) }
							<i>{ __(
								'Convert to Ninja Forms',
								'cf_conversion_tool'
							) }</i>.
						</p>
						<p>
							{ __(
								'To generate an export of this form as a Ninja Forms form export, please select Download as .nff File.',
								'cf_conversion_tool'
							) }{ ' ' }
						</p>
						<div className="cf-converter-buttons">
							<Button
								isPrimary
								onClick={ () =>
									conversionAction( 'create_form' )
								}
								isBusy={ createForm }
								disabled={ downloadFile }
								showTooltip="true"
								label={ __(
									'Make sure Ninja Forms is installed \r\n and activated in order to use\r\n the create Form option.',
									'cf_conversion_tool'
								) }
								text={ __(
									'Convert to Ninja Forms',
									'cf_conversion_tool'
								) }
							/>
							<Button
								isPrimary
								onClick={ () =>
									conversionAction( 'download_file' )
								}
								isBusy={ downloadFile }
								disabled={ createForm }
								text={ __(
									'Download as .nff File',
									'cf_conversion_tool'
								) }
							/>
							<Button
								isSecondary
								onClick={ handleModal }
								text={ __( 'Cancel', 'cf_conversion_tool' ) }
							/>
						</div>
						<div className="cf-converter-response">
							{ spinnerOn && (
								<div>
									<p>
										{ __(
											'Please hold, this could take a while',
											'cf_conversion_tool'
										) }
									</p>
									<Spinner />
									<p>
										{ __(
											'1 - Get CF JSON data: ',
											'cf_conversion_tool'
										) }
										{ getCfFormStep
											? getCfFormStep
											: __(
													'Not triggered',
													'cf_conversion_tool'
											  ) }
									</p>
									<p>
										{ __(
											'2 - Translate to .NFF data: ',
											'cf_conversion_tool'
										) }
										{ getNfFormTranslation
											? getNfFormTranslation
											: __(
													'Not Triggered',
													'cf_conversion_tool'
											  ) }
									</p>
									<p>
										{ createForm &&
											__(
												'3 - Create Form: ',
												'cf_conversion_tool'
											) }
										{ downloadFile &&
											__(
												'3 - Download File: ',
												'cf_conversion_tool'
											) }
										{ getNfFormCreationProcess
											? getNfFormCreationProcess.message
											: __(
													'Not Triggered',
													'cf_conversion_tool'
											  ) }
									</p>
								</div>
							) }
							{ getFormResponse && (
								<div>
									<p>{ getFormResponse.message }</p>
									{ getFormResponse.action ===
										'create_form' && (
										<a
											href={
												cf_conversion_tool_vars.admin_url +
												'?page=ninja-forms&form_id=' +
												getFormResponse.form
											}
											title={ __(
												'Visit new Ninja Forms edit Page',
												'cf_conversion_tool'
											) }
										>
											{ __(
												'Go to Ninja Forms Edit page',
												'cf_conversion_tool'
											) }
										</a>
									) }
								</div>
							) }
							<br />
							{ getFormResponse.code === 'fetch_error' &&
								__(
									'Please refresh the page to initiate new Form conversion.',
									'cf_conversion_tool'
								) }
						</div>
					</Modal>
				) }
				{ spinnerOn && ! modalOpen && (
					<Modal
						className="cf-converter-alert-modal"
						title={ __(
							'Do you want to abort form conversion?',
							'cf_conversion_tool'
						) }
						onRequestClose={ handleModal }
					>
						<div className="cf-converter-alert-buttons">
							<Button
								isDestructive
								className="cf-converter-cancel-action-button"
								text={ __( 'Yes', 'cf_conversion_tool' ) }
								onClick={ cancelAction }
							/>
							<Button
								isSecondary
								onClick={ handleModal }
								text={ __( 'No', 'cf_conversion_tool' ) }
							/>
						</div>
					</Modal>
				) }
			</>
		);
	}
}

ModalWindow.propTypes = {
	formID: PropTypes.string.isRequired,
	form: PropTypes.object,
};
