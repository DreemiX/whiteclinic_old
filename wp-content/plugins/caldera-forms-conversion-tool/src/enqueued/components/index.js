import { ModalWindow } from './modalWindow.js';
import { fetchForm } from './fetchForm.js';
import { callConversion } from './callConversion.js';
import { downloadAsNff } from './downloadAsNff.js';
import { createNfForm } from './createNfForm.js';
export { ModalWindow, fetchForm, callConversion, downloadAsNff, createNfForm };
