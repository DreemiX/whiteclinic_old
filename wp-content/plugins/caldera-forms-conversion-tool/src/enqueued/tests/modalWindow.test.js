import { render, fireEvent, screen } from '@testing-library/react';
import { withGlobal, cf_conversion_tool_vars } from './mockData';
import { ModalWindow } from '../components/';

describe( 'ModalWindow', () => {
	it( 'Matches snapshot', () => {
		//Start render with global wrappers
		withGlobal( 'cf_conversion_tool_vars', cf_conversion_tool_vars, () => {
			const { container } = render( <ModalWindow formID="cf-1" form={ cf_conversion_tool_vars.forms["cf-1"] }/> );
			expect( container ).toMatchSnapshot();
		} );
	} );

	it( 'Opens modal with correct CF ID', () => {
		//Start render with global wrappers
		withGlobal( 'cf_conversion_tool_vars', cf_conversion_tool_vars, () => {
			const { container } = render( <ModalWindow formID="cf-2" form={ cf_conversion_tool_vars.forms["cf-2"] } /> );

			fireEvent.click( screen.getByText( 'Convert to Ninja Forms' ) );
			expect(
				screen.getByText( 'Convert form cf-2 into a Ninja Forms Form.' )
			).toBeDefined();
		} );
	} );
} );
