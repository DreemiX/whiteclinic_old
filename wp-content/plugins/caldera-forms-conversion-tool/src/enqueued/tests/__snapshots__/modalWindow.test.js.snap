// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWindow Matches snapshot 1`] = `
<div>
   
  |
  <button
    class="components-button cf-converter-modal-action is-tertiary"
    type="button"
  >
    Convert to Ninja Forms
  </button>
</div>
`;
